packages/datetime/src/dateRangeInput.tsx

```diff
@@ -148,6 +196,23 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         this.setState({ isOpen: false });
     }
 
+    private getInitialRange = (props = this.props): moment.Moment[] => {
+        const { defaultValue, value } = props;
+
+        let initialStart: moment.Moment;
+        let initialEnd: moment.Moment;
+
+        if (value != null) {
+            [initialStart, initialEnd] = fromDateRangeToMomentArray(value);
+        } else if (defaultValue != null) {
+            [initialStart, initialEnd] = fromDateRangeToMomentArray(defaultValue);
+        } else {
+            [initialStart, initialEnd] = [moment(null), moment(null)];
```

Comment:
I generally prefer having a single `return` at the end of a function, but yeah: returning rather than assigning gets rid of the mutability here, which is nicer. 👍